arrow/arrow.py

```diff
@@ -1390,84 +1496,140 @@ def for_json(self):
 
     # math
 
-    def __add__(self, other):
+    @overload
+    def __add__(self, other: timedelta) -> "Arrow":
+        pass  # pragma: no cover
+
+    @overload
+    def __add__(self, other: relativedelta) -> "Arrow":
+        pass  # pragma: no cover
+
+    def __add__(self, other: Any) -> "Arrow":
```

Comment:
We need overloads here since we need to trigger a `NotImplemented` error in the case of an inappropriate type correct?

Reply:
Yeap. if I remove overloads and set `other` as `Union[timedelta, relativedelta]`, mypy blame that `return NotImplemented` line is unreachable.